server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java

```diff
@@ -217,9 +217,13 @@ public void createPendingSegmentsTable(final String tableName)
                 tableName, getPayloadType(), getQuoteString()
             ),
             StringUtils.format(
-                "CREATE INDEX idx_%1$s_datasource_used_end ON %1$s(dataSource, used, %2$send%2$s)",
+                "CREATE INDEX idx_%1$s_datasource_end ON %1$s(dataSource, %2$send%2$s)",
```

Comment:
I wonder it's better to make an index for `dataSource, sequence_name, end` because this combination is used when searching an existing segmentId in `IndexerSQLMetadataStorageCoordinator.allocatePendingSegment()`.

Reply:
We generally expect a small number of pending segments per sequence name, so I didn't think adding `end` would help much.